### ISSUE TITLE
Discrete Boundary-Seeking GANs

### ISSUE THREAD

**Issue by ankeshanand** [wiseodd/generative-models#15]
Hi,

I think the current implementation for Boundary-seeking GANs is only applicable for continuous variables, and not for discrete which was the main contribution of the paper. Is a discrete implementation in the works?

**Comment by wiseodd** [comment]
Hi,
No, it's not in the work.

**Comment by AjayTalati** [comment]
Hi guys :)

@ankeshanand, why don't you try the current implementation on binarized MNIST? 

I think you only need to use `torch.round` on `X`, in this line

https://github.com/wiseodd/generative-models/blob/master/GAN/boundary_seeking_gan/bgan_pytorch.py#L57

Why not give it a try?

**Comment by ankeshanand** [comment]
@AjayTalati 

Don't think it's that simple. If you go through the paper, the discretized version is different from the continuous version. The gradients are computed analytically there, not via backprop.

**Comment by AjayTalati** [comment]
@ankeshanand thanks, I had a look. 

Luckily PyTorch has an easy to implement REINFORCE method. Perhaps you might want to use that to get a simple baseline, so at least you can reproduce some of their paper? 

If you did that, then you could implement their method, and compare the two.

**Comment by rdevon** [comment]
Hi there. It's actually quite easy to implement discrete BGAN using standard backprop methods. 
Lasagne implementation:
Binary variables:
https://github.com/rdevon/BGAN/blob/master/lib/loss.py#L176
General multinomial:
https://github.com/rdevon/BGAN/blob/master/lib/loss.py#L230

I'm sure it would be easy to implement in PyTorch
